Add digits

taylorskalyo opened the pull request:
![screenshot_20220814_143532](https://user-images.githubusercontent.com/2508427/184550520-0b306d89-8c4c-4f92-8a62-4dd71f6f347d.png)
![screenshot_20220814_143704](https://user-images.githubusercontent.com/2508427/184550521-fee86e46-0006-4cce-8122-2eb5cc2b8403.png)
![screenshot_20220814_143834](https://user-images.githubusercontent.com/2508427/184550523-a17e6889-a1a1-40bd-8736-d71c1161ffe1.png)

taylorskalyo committed:
Use texture names to build letter tables

taylorskalyo committed:
Consolidate lower/upper functions

taylorskalyo committed:
Add support for digits

SmallJoker commented:
The code simplification is great to see. Would you please be so nice to run `optipng -strip all` or `pngcrush` on the added textures? That will speed up loading time of clients and lower the data throughput for servers.

taylorskalyo committed:
Optimize image size with optipng -strip all

taylorskalyo commented:
Good call! That made the new images much smaller.

SmallJoker commented on init.lua:
```diff
@@ -64,203 +49,73 @@ function letters.register_letters(modname, subname, from_node, description, tile
+	for _, tile in ipairs(minetest.get_dir_list(letterspath .. "/textures", false)) do
+		local char, group = tile:match("_([%d%u%l])(%l)_overlay")
+		if char and group then
+			def = table.copy(def)
```
This is prone to issues caused by read access to `def`, which was changed by the previous iteration. Please use a new variable here to always begin with the same "template" table.

taylorskalyo replied:
Can't believe I missed that. Thank you!

SmallJoker commented on init.lua:
```diff
@@ -64,203 +49,73 @@ function letters.register_letters(modname, subname, from_node, description, tile
-end
-
-function letter_cutter:get_output_inv_upper(modname, subname, amount, max)
+function letter_cutter:get_output_inv(modname, subname, amount, max, group)
```
While you're at it, would you please be so nice to add comments where it makes sense? For example, what this function and its parameters do.
EDIT: Same for the other new code. I very much like the generic approach, but would welcome comments too.

SmallJoker reviewed (commented):
:+1:  Works. Few comments above ^

taylorskalyo committed:
Create a new def each iteration from the base def

taylorskalyo committed:
Add comments

- Add LuaDoc comments for public functions
- Add short description for private functions
- Add comments for recent code additions

taylorskalyo committed:
Enable unused arg checking

taylorskalyo committed:
Use maintained luacheck

Roang-zero1/factorio-mod-luacheck seems to be using
https://github.com/mpeterv/luacheck, which hasn't been touched since
2018. Lunar Modules maintains an active fork.
